packages/frontend-2/components/settings/server/user/InviteDialog.vue

```diff
@@ -1,119 +1,161 @@
+    <template #header>Invite to server</template>
     <form @submit="onSubmit">
-      <div class="flex flex-col gap-y-3 text-foreground mb-4">
-        <p class="text-body-xs">
-          Speckle will send a server invite link to the email(-s) below. You can also
-          add a personal message if you want to. To add multiple e-mails, seperate them
-          with commas.
-        </p>
-        <FormTextInput
-          name="emailsString"
-          color="foundation"
-          label="E-mail"
-          show-label
-          placeholder="example@example.com, example2@example.com"
-          :rules="[isRequired, isOneOrMultipleEmails]"
-          :disabled="anyMutationsLoading"
-        />
-        <FormSelectServerRoles
-          v-if="allowServerRoleSelect"
-          v-model="serverRole"
-          label="Select server role"
-          class="w-full sm:w-60"
-          show-label
-          :allow-guest="isGuestMode"
-          :allow-admin="isAdmin"
-          mount-menu-on-body
-        />
-        <FormTextArea
-          name="message"
-          show-optional
-          show-label
-          label="Message"
-          color="foundation"
+      <div class="flex flex-col gap-y-5 text-foreground mb-4">
+        <div v-for="(item, index) in fields" :key="item.key" class="flex gap-x-3">
+          <div class="flex flex-col gap-y-3 flex-1">
+            <hr v-if="index !== 0" class="border-outline-3" />
+            <div class="flex flex-row gap-x-3">
+              <div class="flex-1">
+                <FormTextInput
+                  :name="`email-${index}`"
+                  color="foundation"
+                  label="E-mail"
+                  show-label
+                  placeholder="Enter an email to invite"
+                  :rules="[isRequired, isEmail]"
+                  :disabled="anyMutationsLoading"
+                  @input="
+                    (e) =>
+                      updateField({ field: InputFields.Email, val: e.value, index })
+                  "
+                />
+              </div>
+              <FormSelectServerRoles
+                v-if="allowServerRoleSelect"
+                label="Select role"
+                class="sm:w-48"
+                show-label
+                :disabled="anyMutationsLoading"
+                :allow-guest="isGuestMode"
+                :allow-admin="isAdmin"
+                :name="`role-${index}`"
+                :rules="[isRequired]"
+                mount-menu-on-body
+                @update:model-value="(e) => onUpdateRole(e, index)"
+              />
+            </div>
+            <FormSelectProjects
+              label="Select project"
+              class="w-full"
+              owned-only
+              show-optional
+              mount-menu-on-body
+              show-label
+              :name="`project-${index}`"
+              @update:model-value="(e) => onUpdateProject(e, index)"
+            />
+          </div>
+          <div class="relative w-4">
```

Comment:
v-if on CommonTextLink leaves the parent div taking up space
<img width="600" alt="image" src="https://github.com/user-attachments/assets/50660afe-2d40-4934-8450-3d0ed3481825">

Reply:
Maybe this was intentional to stop a layout shift when the icon comes in?

Reply:
It's indeed intended according to the design :)